### DISCUSSION TITLE
What is being done to prevent bad words or hurtful strings?

### DISCUSSION THREAD

**Discussion by nicklenko** [faker-js/faker#1864] [Q&A]
Hi there, we're looking to use this in a customer facing previews and want to be sure that there are no bad text strings being generated. Is there a way to view the list of strings that are used to generate the fake info? Or are there other methods used to ensure no offensive strings are generated?

**Comment by Shinigami92** [comment]
We did stuff like this
- https://github.com/faker-js/faker/pull/1745

**Comment by ST-DDT** [comment]
>  we're looking to use this in a customer facing previews and want to be sure that there are no bad text strings being generated.

First of all there is no guarantee that there won't be any bad words in the results.
Since we took over the project we have removed words, that are or can be considered offensive.

- https://github.com/faker-js/faker/pull/1633 ([v8.0.0-alpha.0](https://github.com/faker-js/faker/releases/tag/v8.0.0-alpha.0))
- https://github.com/faker-js/faker/pull/1745 (not released yet)

If you find more, please report them to us, so we can remove them.

> Is there a way to view the list of strings that are used to generate the fake info?

Yes, you can review the locale data that will be returned in the methods:

https://github.com/faker-js/faker/tree/next/src/locales

> Or are there other methods used to ensure no offensive strings are generated?

We ran offensive word filters on the existing locales and do this for new locale data as well, but the tools don't support all locales and don't find all words (especially if they contain typos). We also face the issue, that some words are considered offensive in some locales, but not it others.
We appreciate any help with reviewing the locales.

Note: The string module generates random texts from characters, so with lots of bad luck it might generate offensive content.

If you want to be 100% sure that nothing that could be considered offensive for anyone is returned, you have to apply manual filtering.

**Comment by matthewmayer** [comment]
Also be aware that if this is customer facing emails, phones, addresses etc might coincidentally be the same as real emails, phones, addresses.

**Reply by Johnkinging** [reply]
Are the emails not all `@example.com`? Just curious, that's all I've seen in practice.

**Reply by ST-DDT** [reply]
There is [faker.internet.email](https://fakerjs.dev/api/internet.html#email) that generates non example ones.
And [faker.internet.exampleEmail](https://fakerjs.dev/api/internet.html#exampleemail) that generates example ones.

**Reply by Johnkinging** [reply]
Ah, good to know, thanks!

**Comment by mcavaliere** [comment]
It appears that some of the words that aren't in the raw source are still appearing in the generated cjs/esm modules. 

I found this output (from `faker.internet.url` which traces down to `noun()`

<img width="730" alt="image" src="https://user-images.githubusercontent.com/147237/220933901-c23aedc3-7fb5-47bc-ad2f-3622b961d697.png">

The word `cock` isn't in the noun word list [here](https://github.com/faker-js/faker/blob/next/src/locales/en/word/noun.ts), but it's in my generated output in `node_modules` quite a few times: 

![image](https://user-images.githubusercontent.com/147237/220934284-402189f6-e074-486e-b442-becb12552261.png)

**Reply by mcavaliere** [reply]
Does this just need another publish?

**Reply by Shinigami92** [reply]
are you already on v8.0.0-alpha.0?

**Reply by ST-DDT** [reply]
The second more thorough cleanup isn't released yet: https://github.com/faker-js/faker/discussions/1864#discussioncomment-5068196

**Comment by mcavaliere** [comment]
I am now, thanks!

On Thu, Feb 23, 2023 at 5:49 PM ST-DDT ***@***.***> wrote:

> The second more thorough cleanup isn't released yet.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/faker-js/faker/discussions/1864#discussioncomment-5094551>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABD6JOVWUTQL73TCRDJ7ODWY7SP5ANCNFSM6AAAAAAVDMIU3Y>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 

*Peace.**Mike Cavaliere*
Software Strategist / Author
*cutintothejamstack.com <http://cutintothejamstack.com/>*
<http://www.linkedin.com/in/mikecavaliere>
linkedin.com/in/mikecavaliere <http://www.linkedin.com/in/mikecavaliere>
@mcavaliere <http://twitter.com/mcavaliere>
